shell/activate.sh

```diff
@@ -174,23 +190,33 @@ fi
 # store remaining values that may get cleared by deactivate               #
 _CONDA_WHAT_SHELL_AM_I="${WHAT_SHELL_AM_I}"
 _CONDA_VERBOSE="${CONDA_VERBOSE}"
+_CONDA_STACK="${CONDA_STACK}"
 _IS_ENV_CONDA_VERBOSE="${IS_ENV_CONDA_VERBOSE}"
+_IS_ENV_CONDA_STACK="${IS_ENV_CONDA_STACK}"
 if [ -n "${MSYS2_ENV_CONV_EXCL+x}" ]; then
     _MSYS2_ENV_CONV_EXCL="${MSYS2_ENV_CONV_EXCL}"
 fi
 
 # ensure we deactivate any scripts from the old env                       #
-. deactivate.sh ""
-if [ $? != 0 ]; then
-    unset _CONDA_WHAT_SHELL_AM_I
-    [ "${IS_ENV_CONDA_ENVNAME}" = "0" ] && unset CONDA_ENVNAME
-    unset _CONDA_VERBOSE
-    unset IS_ENV_CONDA_ENVNAME
-    unset _IS_ENV_CONDA_VERBOSE
-    if [ -n "${_MSYS2_ENV_CONV_EXCL+x}" ]; then
-        unset _MSYS2_ENV_CONV_EXCL
+if [ -n "${CONDA_STACK+x}" ]; then
+    . deactivate.sh ""
+    if [ $? != 0 ]; then
+        unset _CONDA_WHAT_SHELL_AM_I
+        [ "${IS_ENV_CONDA_ENVNAME}" = "0" ] && unset CONDA_ENVNAME
+        unset _CONDA_VERBOSE
+        unset _CONDA_STACK
+        unset IS_ENV_CONDA_ENVNAME
+        unset _IS_ENV_CONDA_VERBOSE
+        unset _IS_ENV_CONDA_STACK
+        if [ -n "${_MSYS2_ENV_CONV_EXCL+x}" ]; then
+            unset _MSYS2_ENV_CONV_EXCL
+        fi
+        return 1
+    fi
+else
+    if [ "${CONDA_VERBOSE}" = "${TRUE}" ]; then
+        echo "[ACTIVATE]: Warning: stacking environments (--stacks option) can cause unexpected errors in case of incompatibilities between different conda packages"
     fi
-    return 1
```

Comment:
This effectively means that any --stack will detect that it's being stacked and then not deactivate the current environment and then exits without activating the new environment to stack.